src/app/models/AbstractModel.spec.ts

```diff
@@ -182,6 +182,28 @@ describe("AbstractModel", () => {
           expect(hasFormDataOnlyAttributes(model)).toBeFalse();
         });
 
+        it("should return true if only attribute is a File type object", () => {
+          const testFile = new File([""], "testFileName.png");
+          class Model extends MockModel {
+            @bawPersistAttr({ supportedFormats: ["formData"] })
+            public value0: any;
+          }
+          const model = new Model({value0: testFile});
+          expect(hasFormDataOnlyAttributes(model)).toBeTrue();
+        });
+
+        it("should return true if an attribute is a File type object", () => {
+          const testFile = new File([""], "testFileName.png");
+          class Model extends MockModel {
+            @bawPersistAttr({ supportedFormats: ["formData"] })
+            public value0: any;
+            @bawPersistAttr({ supportedFormats: ["formData"] })
+            public value1: any;
+          }
+          const model = new Model({value0: testFile, value1: "text input"});
+          expect(hasFormDataOnlyAttributes(model)).toBeTrue();
+        });
+
```

Comment:
- you didn't test the null state
- you didn't test the other two json serialization functions

Reply:
The null state check was added in the previous commit (c4dd9597ae1c1c02aa9efa7fb2b4272a084260da):
 ```
       it("should return false if only attribute is null", () => {
          class Model extends MockModel {
            @bawPersistAttr({ supportedFormats: ["formData"] })
            public value0: any;
          }
          const model = new Model({value0: null});
          expect(hasFormDataOnlyAttributes(model)).toBeFalse();
        });
```